planning/behavior_path_planner/include/behavior_path_planner/scene_module_v2/scene_module_interface.hpp

```diff
@@ -0,0 +1,298 @@
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->publishCooperateStatus(clock_->now());
+  }
+
+  /**
+   * @brief Return true if the activation command is received
+   */
+  virtual bool isActivated()
+  {
+    if (!rtc_interface_ptr_) {
+      return true;
+    }
+    if (rtc_interface_ptr_->isRegistered(uuid_)) {
+      return rtc_interface_ptr_->isActivated(uuid_);
+    }
+    return false;
+  }
+
+  virtual void publishSteeringFactor()
+  {
+    if (!steering_factor_interface_ptr_) {
+      return;
+    }
+    steering_factor_interface_ptr_->publishSteeringFactor(clock_->now());
+  }
+
+  virtual void lockRTCCommand()
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->lockCommandUpdate();
+  }
+
+  virtual void unlockRTCCommand()
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->unlockCommandUpdate();
+  }
+
+  /**
+   * @brief set previous module's output as input for this module
+   */
+  void setPreviousModuleOutput(const BehaviorModuleOutput & previous_module_output)
+  {
+    previous_module_output_ = previous_module_output;
+  }
+
+  /**
+   * @brief set planner data
+   */
+  void setData(const std::shared_ptr<const PlannerData> & data) { planner_data_ = data; }
+
+  void resetPathCandidate() { path_candidate_.reset(); }
+
+  void resetPathReference() { path_reference_.reset(); }
+
+  bool isWaitingApproval() const { return is_waiting_approval_; }
+
+  bool isAlreadyApproved() const { return is_already_approved_; }
+
+  rclcpp::Logger getLogger() const { return logger_; }
+
+  PlanResult getPathCandidate() const { return path_candidate_; }
+
+  PlanResult getPathReference() const { return path_reference_; }
+
+  MarkerArray getDebugMarkers() { return debug_marker_; }
+
+  ModuleStatus getCurrentStatus() const { return current_state_; }
+
+  virtual void acceptVisitor(const std::shared_ptr<SceneModuleVisitor> & visitor) const = 0;
+
+  std::string name() const { return name_; }
+
+  std::shared_ptr<const PlannerData> planner_data_;
+
+private:
+  std::string name_;
+  rclcpp::Logger logger_;
+
+protected:
+  void updateRTCStatus(const double start_distance, const double finish_distance)
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->updateCooperateStatus(
+      uuid_, isExecutionReady(), start_distance, finish_distance, clock_->now());
+  }
+
+  virtual void removeRTCStatus()
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->clearCooperateStatus();
+  }
+
+  void alreadyApproved() { is_already_approved_ = true; }
+
+  void nothingApproved() { is_already_approved_ = false; }
+
+  void waitApproval() { is_waiting_approval_ = true; }
+
+  void clearWaitingApproval() { is_waiting_approval_ = false; }
+
+  rclcpp::Clock::SharedPtr clock_;
+
+  std::shared_ptr<RTCInterface> rtc_interface_ptr_;
+  std::unique_ptr<SteeringFactorInterface> steering_factor_interface_ptr_;
```

Comment:
`rtc_interface` and `steering_factor_interface` have a publisher, so `SceneModuleManager` should have these interfaces, not `SceneModuleInterface`.

The implementation in `behavior_velocity_planner` may be helpful.
https://github.com/autowarefoundation/autoware.universe/blob/b28d792dee2379b7e9561ec256dc0052c60d127b/planning/behavior_velocity_planner/include/scene_module/scene_module_interface.hpp#L345-L430

(There is no problem if only one scene module is registered. If it is hard to change, it is okay to do it later.)